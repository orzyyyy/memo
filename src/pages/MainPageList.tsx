import React from 'react';
import { format } from 'date-fns';
import { MappingProps } from '../../server/controller/DocumentController';
import { DocumentCategoryProps } from '../../server/utils/document';

const MainPageList = ({
  onDelete,
  onEdit,
  onHide,
  siderSelectedKey,
  dataSource,
  onListItemClick,
  isLocal,
}: {
  siderSelectedKey: string;
  onDelete?: (dataItem: MappingProps) => void;
  onHide?: (dataItem: MappingProps) => void;
  dataSource: MappingProps[];
  onEdit: (dataItem?: MappingProps, visible?: boolean, pageInfo?: { x: number; y: number }) => void;
  onListItemClick: ({ category, id }: { category: DocumentCategoryProps; id: string }) => void;
  isLocal: boolean;
}) => {
  const data = siderSelectedKey === 'all' ? dataSource : dataSource.filter(item => item.subType === siderSelectedKey);

  return (
    <ul>
      {data.map(item => {
        const buttonGroup = (
          <div style={{ float: 'right', marginRight: 8 }}>
            <a
              onClick={(e: React.MouseEvent) => {
                e.stopPropagation();
                onEdit(item, true);
              }}
              style={{ marginRight: 16 }}
            >
              修改
            </a>
            <a
              onClick={(e: React.MouseEvent) => {
                e.stopPropagation();
                onHide && onHide(item);
              }}
              style={{ marginRight: 16 }}
            >
              隐藏
            </a>
            <a
              onClick={(e: React.MouseEvent) => {
                e.stopPropagation();
                onDelete && onDelete(item);
              }}
              style={{ marginRight: 16 }}
            >
              删除
            </a>
          </div>
        );

        return (
          <li
            key={`list-item-${item.id}`}
            className="list-item"
            onClick={() => onListItemClick({ category: item.category, id: item.id })}
          >
            {item.category === 'mapping' && <div style={{ background: '#108ee9' }} className="icon-apartment" />}
            {item.category === 'markdown' && <div style={{ background: '#87d068' }} className="icon-file-markdown" />}
            {item.type + ' - ' + item.subType + ' - ' + item.title}
            <div style={{ float: 'right', marginRight: 8 }}>
              {`${format(new Date(item.createTime || ''), 'yyyy-MM-dd')} / ${format(
                new Date(item.modifyTime || ''),
                'yyyy-MM-dd',
              )}`}
            </div>
            {isLocal && buttonGroup}
          </li>
        );
      })}
    </ul>
  );
};

export default MainPageList;
