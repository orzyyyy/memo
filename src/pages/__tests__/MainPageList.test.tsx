import React from 'react';
import { mount } from 'enzyme';
import MainPageList from '../MainPageList';

const dataSource = [
  {
    id: '32e89f38116080d6b7608cf2137f3c96',
    title: '音程表',
    url: './assets/mapping/32e89f38116080d6b7608cf2137f3c96.json',
    createTime: 1559522075108,
    modifyTime: 1559523700429,
    type: '音乐',
    subType: '乐理',
    category: undefined,
  },
  {
    id: '7a73d21c13f1eb55053d4af66e0d057b',
    title: 'windows 下安装 rust',
    url: './assets/mapping/7a73d21c13f1eb55053d4af66e0d057b.json',
    createTime: 1559617097664,
    modifyTime: 1559633639037,
    type: 'Rust',
    subType: '安装',
    category: undefined,
  },
];

describe('MainPageList', () => {
  const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

  afterAll(() => {
    errorSpy.mockRestore();
  });

  it('render correctly', () => {
    const onDelete = jest.fn();
    const onEdit = jest.fn();
    const onListItemClick = jest.fn();
    const wrapper = mount(
      <MainPageList
        siderSelectedKey="all"
        dataSource={dataSource}
        onDelete={onDelete}
        onEdit={onEdit}
        onListItemClick={onListItemClick}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('events works correctly', () => {
    const onDelete = jest.fn();
    const onEdit = jest.fn();
    const onListItemClick = jest.fn();
    const wrapper: any = mount(
      <MainPageList
        siderSelectedKey="all"
        dataSource={dataSource}
        onDelete={onDelete}
        onEdit={onEdit}
        onListItemClick={onListItemClick}
      />,
    );
    wrapper
      .find('Item')
      .first()
      .simulate('contextmenu');
    wrapper
      .find('.ant-dropdown-menu-item')
      .first()
      .props()
      .onClick();
    expect(onEdit).toHaveBeenCalled();
    wrapper
      .find('.ant-dropdown-menu-item')
      .last()
      .props()
      .onClick();
    expect(onDelete).toHaveBeenCalled();
    wrapper
      .find('Item')
      .first()
      .simulate('click');
    expect(onListItemClick).toHaveBeenCalled();
  });
});
