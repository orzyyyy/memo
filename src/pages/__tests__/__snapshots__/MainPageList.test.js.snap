// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageList render correctly 1`] = `
<MainPageList
  props={
    Object {
      "dataSource": Array [
        Object {
          "category": "markdown",
          "createTime": 1559522075108,
          "id": "32e89f38116080d6b7608cf2137f3c96",
          "modifyTime": 1559523700429,
          "subType": "乐理",
          "title": "音程表",
          "type": "音乐",
          "url": "./assets/mapping/32e89f38116080d6b7608cf2137f3c96.json",
        },
        Object {
          "category": "mapping",
          "createTime": 1559617097664,
          "id": "7a73d21c13f1eb55053d4af66e0d057b",
          "modifyTime": 1559633639037,
          "subType": "安装",
          "title": "windows 下安装 rust",
          "type": "Rust",
          "url": "./assets/mapping/7a73d21c13f1eb55053d4af66e0d057b.json",
        },
      ],
      "onDelete": [MockFunction],
      "onEdit": [MockFunction],
    }
  }
  state={
    Object {
      "siderSelectedKey": "all",
    }
  }
>
  <List
    bordered={false}
    dataSource={
      Array [
        Object {
          "category": "markdown",
          "createTime": 1559522075108,
          "id": "32e89f38116080d6b7608cf2137f3c96",
          "modifyTime": 1559523700429,
          "subType": "乐理",
          "title": "音程表",
          "type": "音乐",
          "url": "./assets/mapping/32e89f38116080d6b7608cf2137f3c96.json",
        },
        Object {
          "category": "mapping",
          "createTime": 1559617097664,
          "id": "7a73d21c13f1eb55053d4af66e0d057b",
          "modifyTime": 1559633639037,
          "subType": "安装",
          "title": "windows 下安装 rust",
          "type": "Rust",
          "url": "./assets/mapping/7a73d21c13f1eb55053d4af66e0d057b.json",
        },
      ]
    }
    loading={false}
    pagination={false}
    renderItem={[Function]}
    split={true}
  >
    <div
      className="ant-list ant-list-split"
    >
      <Spin
        size="default"
        spinning={false}
        wrapperClassName=""
      >
        <div
          className="ant-spin-nested-loading"
        >
          <div
            className="ant-spin-container"
            key="container"
          >
            <ul
              className="ant-list-items"
            >
              <Dropdown
                key="list-item-0"
                mouseEnterDelay={0.15}
                mouseLeaveDelay={0.1}
                overlay={[Function]}
                placement="bottomLeft"
                trigger={
                  Array [
                    "contextMenu",
                  ]
                }
              >
                <Dropdown
                  alignPoint={true}
                  defaultVisible={false}
                  mouseEnterDelay={0.15}
                  mouseLeaveDelay={0.1}
                  onVisibleChange={[Function]}
                  overlay={[Function]}
                  overlayClassName=""
                  overlayStyle={Object {}}
                  placement="bottomLeft"
                  prefixCls="ant-dropdown"
                  showAction={Array []}
                  transitionName="slide-up"
                  trigger={
                    Array [
                      "contextMenu",
                    ]
                  }
                >
                  <Trigger
                    action={
                      Array [
                        "contextMenu",
                      ]
                    }
                    afterPopupVisibleChange={[Function]}
                    alignPoint={true}
                    blurDelay={0.15}
                    builtinPlacements={
                      Object {
                        "bottomCenter": Object {
                          "offset": Array [
                            0,
                            4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "tc",
                            "bc",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "bottomLeft": Object {
                          "offset": Array [
                            0,
                            4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "tl",
                            "bl",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "bottomRight": Object {
                          "offset": Array [
                            0,
                            4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "tr",
                            "br",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "topCenter": Object {
                          "offset": Array [
                            0,
                            -4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "bc",
                            "tc",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "topLeft": Object {
                          "offset": Array [
                            0,
                            -4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "bl",
                            "tl",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "topRight": Object {
                          "offset": Array [
                            0,
                            -4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "br",
                            "tr",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                      }
                    }
                    defaultPopupVisible={false}
                    defaultVisible={false}
                    destroyPopupOnHide={false}
                    focusDelay={0}
                    getDocument={[Function]}
                    getPopupClassNameFromAlign={[Function]}
                    hideAction={
                      Array [
                        "click",
                      ]
                    }
                    mask={false}
                    maskClosable={true}
                    mouseEnterDelay={0.15}
                    mouseLeaveDelay={0.1}
                    onPopupAlign={[Function]}
                    onPopupVisibleChange={[Function]}
                    onVisibleChange={[Function]}
                    overlay={[Function]}
                    popup={[Function]}
                    popupAlign={Object {}}
                    popupClassName=""
                    popupPlacement="bottomLeft"
                    popupStyle={Object {}}
                    popupTransitionName="slide-up"
                    popupVisible={false}
                    prefixCls="ant-dropdown"
                    showAction={Array []}
                  >
                    <Item
                      className="list-item ant-dropdown-trigger"
                      key="trigger"
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      <li
                        className="ant-list-item list-item ant-dropdown-trigger ant-list-item-no-flex"
                        onClick={[Function]}
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                        onTouchStart={[Function]}
                      >
                        <FileMarkdown
                          style={
                            Object {
                              "color": "#87d068",
                              "fontSize": 16,
                              "marginRight": 10,
                            }
                          }
                        >
                          <AntdIcon
                            icon={
                              Object {
                                "icon": Object {
                                  "attrs": Object {
                                    "focusable": "false",
                                    "viewBox": "64 64 896 896",
                                  },
                                  "children": Array [
                                    Object {
                                      "attrs": Object {
                                        "d": "M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0042 42h216v494zM429 481.2c-1.9-4.4-6.2-7.2-11-7.2h-35c-6.6 0-12 5.4-12 12v272c0 6.6 5.4 12 12 12h27.1c6.6 0 12-5.4 12-12V582.1l66.8 150.2a12 12 0 0011 7.1H524c4.7 0 9-2.8 11-7.1l66.8-150.6V758c0 6.6 5.4 12 12 12H641c6.6 0 12-5.4 12-12V486c0-6.6-5.4-12-12-12h-34.7c-4.8 0-9.1 2.8-11 7.2l-83.1 191-83.2-191z",
                                      },
                                      "tag": "path",
                                    },
                                  ],
                                  "tag": "svg",
                                },
                                "name": "file-markdown",
                                "theme": "outline",
                              }
                            }
                            style={
                              Object {
                                "color": "#87d068",
                                "fontSize": 16,
                                "marginRight": 10,
                              }
                            }
                          >
                            <span
                              aria-label="file-markdown"
                              className="anticon anticon-file-markdown"
                              role="img"
                              style={
                                Object {
                                  "color": "#87d068",
                                  "fontSize": 16,
                                  "marginRight": 10,
                                }
                              }
                            >
                              <IconReact
                                className=""
                                icon={
                                  Object {
                                    "icon": Object {
                                      "attrs": Object {
                                        "focusable": "false",
                                        "viewBox": "64 64 896 896",
                                      },
                                      "children": Array [
                                        Object {
                                          "attrs": Object {
                                            "d": "M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0042 42h216v494zM429 481.2c-1.9-4.4-6.2-7.2-11-7.2h-35c-6.6 0-12 5.4-12 12v272c0 6.6 5.4 12 12 12h27.1c6.6 0 12-5.4 12-12V582.1l66.8 150.2a12 12 0 0011 7.1H524c4.7 0 9-2.8 11-7.1l66.8-150.6V758c0 6.6 5.4 12 12 12H641c6.6 0 12-5.4 12-12V486c0-6.6-5.4-12-12-12h-34.7c-4.8 0-9.1 2.8-11 7.2l-83.1 191-83.2-191z",
                                          },
                                          "tag": "path",
                                        },
                                      ],
                                      "tag": "svg",
                                    },
                                    "name": "file-markdown",
                                    "theme": "outline",
                                  }
                                }
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="file-markdown"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  key="svg-file-markdown"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0042 42h216v494zM429 481.2c-1.9-4.4-6.2-7.2-11-7.2h-35c-6.6 0-12 5.4-12 12v272c0 6.6 5.4 12 12 12h27.1c6.6 0 12-5.4 12-12V582.1l66.8 150.2a12 12 0 0011 7.1H524c4.7 0 9-2.8 11-7.1l66.8-150.6V758c0 6.6 5.4 12 12 12H641c6.6 0 12-5.4 12-12V486c0-6.6-5.4-12-12-12h-34.7c-4.8 0-9.1 2.8-11 7.2l-83.1 191-83.2-191z"
                                    key="svg-file-markdown-svg-0"
                                  />
                                </svg>
                              </IconReact>
                            </span>
                          </AntdIcon>
                        </FileMarkdown>
                        音乐 - 乐理 - 音程表
                        <div
                          style={
                            Object {
                              "float": "right",
                              "marginRight": 8,
                            }
                          }
                        >
                          2019-06-03 / 2019-06-03
                        </div>
                      </li>
                    </Item>
                  </Trigger>
                </Dropdown>
              </Dropdown>
              <Dropdown
                key="list-item-1"
                mouseEnterDelay={0.15}
                mouseLeaveDelay={0.1}
                overlay={[Function]}
                placement="bottomLeft"
                trigger={
                  Array [
                    "contextMenu",
                  ]
                }
              >
                <Dropdown
                  alignPoint={true}
                  defaultVisible={false}
                  mouseEnterDelay={0.15}
                  mouseLeaveDelay={0.1}
                  onVisibleChange={[Function]}
                  overlay={[Function]}
                  overlayClassName=""
                  overlayStyle={Object {}}
                  placement="bottomLeft"
                  prefixCls="ant-dropdown"
                  showAction={Array []}
                  transitionName="slide-up"
                  trigger={
                    Array [
                      "contextMenu",
                    ]
                  }
                >
                  <Trigger
                    action={
                      Array [
                        "contextMenu",
                      ]
                    }
                    afterPopupVisibleChange={[Function]}
                    alignPoint={true}
                    blurDelay={0.15}
                    builtinPlacements={
                      Object {
                        "bottomCenter": Object {
                          "offset": Array [
                            0,
                            4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "tc",
                            "bc",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "bottomLeft": Object {
                          "offset": Array [
                            0,
                            4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "tl",
                            "bl",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "bottomRight": Object {
                          "offset": Array [
                            0,
                            4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "tr",
                            "br",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "topCenter": Object {
                          "offset": Array [
                            0,
                            -4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "bc",
                            "tc",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "topLeft": Object {
                          "offset": Array [
                            0,
                            -4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "bl",
                            "tl",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                        "topRight": Object {
                          "offset": Array [
                            0,
                            -4,
                          ],
                          "overflow": Object {
                            "adjustX": 1,
                            "adjustY": 1,
                          },
                          "points": Array [
                            "br",
                            "tr",
                          ],
                          "targetOffset": Array [
                            0,
                            0,
                          ],
                        },
                      }
                    }
                    defaultPopupVisible={false}
                    defaultVisible={false}
                    destroyPopupOnHide={false}
                    focusDelay={0}
                    getDocument={[Function]}
                    getPopupClassNameFromAlign={[Function]}
                    hideAction={
                      Array [
                        "click",
                      ]
                    }
                    mask={false}
                    maskClosable={true}
                    mouseEnterDelay={0.15}
                    mouseLeaveDelay={0.1}
                    onPopupAlign={[Function]}
                    onPopupVisibleChange={[Function]}
                    onVisibleChange={[Function]}
                    overlay={[Function]}
                    popup={[Function]}
                    popupAlign={Object {}}
                    popupClassName=""
                    popupPlacement="bottomLeft"
                    popupStyle={Object {}}
                    popupTransitionName="slide-up"
                    popupVisible={false}
                    prefixCls="ant-dropdown"
                    showAction={Array []}
                  >
                    <Item
                      className="list-item ant-dropdown-trigger"
                      key="trigger"
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      <li
                        className="ant-list-item list-item ant-dropdown-trigger ant-list-item-no-flex"
                        onClick={[Function]}
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                        onTouchStart={[Function]}
                      >
                        <Apartment
                          style={
                            Object {
                              "color": "#108ee9",
                              "fontSize": 16,
                              "marginRight": 10,
                            }
                          }
                        >
                          <AntdIcon
                            icon={
                              Object {
                                "icon": Object {
                                  "attrs": Object {
                                    "focusable": "false",
                                    "viewBox": "64 64 896 896",
                                  },
                                  "children": Array [
                                    Object {
                                      "attrs": Object {
                                        "d": "M908 640H804V488c0-4.4-3.6-8-8-8H548v-96h108c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16H368c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h108v96H228c-4.4 0-8 3.6-8 8v152H116c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16H292v-88h440v88H620c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16zm-564 76v168H176V716h168zm84-408V140h168v168H428zm420 576H680V716h168v168z",
                                      },
                                      "tag": "path",
                                    },
                                  ],
                                  "tag": "svg",
                                },
                                "name": "apartment",
                                "theme": "outline",
                              }
                            }
                            style={
                              Object {
                                "color": "#108ee9",
                                "fontSize": 16,
                                "marginRight": 10,
                              }
                            }
                          >
                            <span
                              aria-label="apartment"
                              className="anticon anticon-apartment"
                              role="img"
                              style={
                                Object {
                                  "color": "#108ee9",
                                  "fontSize": 16,
                                  "marginRight": 10,
                                }
                              }
                            >
                              <IconReact
                                className=""
                                icon={
                                  Object {
                                    "icon": Object {
                                      "attrs": Object {
                                        "focusable": "false",
                                        "viewBox": "64 64 896 896",
                                      },
                                      "children": Array [
                                        Object {
                                          "attrs": Object {
                                            "d": "M908 640H804V488c0-4.4-3.6-8-8-8H548v-96h108c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16H368c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h108v96H228c-4.4 0-8 3.6-8 8v152H116c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16H292v-88h440v88H620c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16zm-564 76v168H176V716h168zm84-408V140h168v168H428zm420 576H680V716h168v168z",
                                          },
                                          "tag": "path",
                                        },
                                      ],
                                      "tag": "svg",
                                    },
                                    "name": "apartment",
                                    "theme": "outline",
                                  }
                                }
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="apartment"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  key="svg-apartment"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M908 640H804V488c0-4.4-3.6-8-8-8H548v-96h108c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16H368c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h108v96H228c-4.4 0-8 3.6-8 8v152H116c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16H292v-88h440v88H620c-8.8 0-16 7.2-16 16v288c0 8.8 7.2 16 16 16h288c8.8 0 16-7.2 16-16V656c0-8.8-7.2-16-16-16zm-564 76v168H176V716h168zm84-408V140h168v168H428zm420 576H680V716h168v168z"
                                    key="svg-apartment-svg-0"
                                  />
                                </svg>
                              </IconReact>
                            </span>
                          </AntdIcon>
                        </Apartment>
                        Rust - 安装 - windows 下安装 rust
                        <div
                          style={
                            Object {
                              "float": "right",
                              "marginRight": 8,
                            }
                          }
                        >
                          2019-06-04 / 2019-06-04
                        </div>
                      </li>
                    </Item>
                  </Trigger>
                </Dropdown>
              </Dropdown>
            </ul>
          </div>
        </div>
      </Spin>
    </div>
  </List>
</MainPageList>
`;
